src/dataflow-types/src/reconciliation/command.rs

```diff
@@ -0,0 +1,528 @@
+                    } else {
+                        changes.clear();
+                    }
+                }
+
+                self.responses.push_back(Response::Compute(
+                    ComputeResponse::FrontierUppers(list),
+                    instance,
+                ));
+            }
+            Response::Compute(ComputeResponse::PeekResponse(uuid, response), instance) => {
+                if self.peeks.remove(&uuid) {
+                    self.responses.push_back(Response::Compute(
+                        ComputeResponse::PeekResponse(uuid, response),
+                        instance,
+                    ));
+                }
+            }
+            Response::Compute(ComputeResponse::TailResponse(id, response), instance) => {
+                self.responses.push_back(Response::Compute(
+                    ComputeResponse::TailResponse(id, response),
+                    instance,
+                ));
+            }
+            Response::Storage(_) => {
+                panic!("ComputeCommandReconcile cannot handle Storage responses")
+            }
+        }
+    }
+
+    async fn absorb_compute_command(
+        &mut self,
+        command: ComputeCommand,
+        instance: ComputeInstanceId,
+    ) -> Result<(), anyhow::Error> {
+        use Command::Compute;
+        use ComputeCommand::*;
+        match command {
+            CreateInstance(config) => {
+                // TODO: Handle `logging` correctly when reconnecting. We currently assume that the
+                // logging config stays the same.
+                if self.created.insert(instance) {
+                    if let Some(logging) = &config {
+                        for id in logging.log_identifiers() {
+                            if !self.uppers.contains_key(&(id, instance)) {
+                                self.start_tracking(id, instance);
+                            }
+                        }
+                    }
+                    self.client
+                        .send(Compute(CreateInstance(config), instance))
+                        .await?;
+                }
+                Ok(())
+            }
+            cmd @ DropInstance => {
+                if self.created.remove(&instance) {
+                    self.uppers.retain(|(_, i), _| i != &instance);
+                    self.client.send(Compute(cmd, instance)).await
+                } else {
+                    Ok(())
+                }
+            }
+            CreateDataflows(dataflows) => {
+                let mut create = Vec::new();
+                for dataflow in dataflows {
+                    let mut exports = dataflow
+                        .sink_exports
+                        .iter()
+                        .map(|(id, _)| *id)
+                        .chain(dataflow.index_exports.iter().map(|(id, _, _)| *id))
+                        .collect::<Vec<_>>();
+                    exports.sort_unstable();
+                    exports.dedup();
```

Comment:
We should just have this be a method on `DataflowDescription`, and use it for the (temporary) name as well.